adapters/sigfox-vertx/src/main/java/org/eclipse/hono/adapter/sigfox/impl/SigfoxProtocolAdapter.java

```diff
@@ -185,4 +199,95 @@ private static Buffer decodeData(final String data) {
         return Buffer.buffer(BaseEncoding.base16().decode(data.toUpperCase()));
     }
 
+    @Override
+    protected Integer getTimeUntilDisconnectFromRequest(final RoutingContext ctx) {
+
+        String ack = ctx.queryParams().get(SIGFOX_QUERY_PARAM_ACK);
+
+        if (ack == null) {
+            ack = ctx.request().getHeader(SIGFOX_HEADER_PARAM_ACK);
+        }
+
+        if (ack == null) {
+            return null;
+        }
+
+        LOG.debug("Ack required for request?: '{}'", ack);
+
+        if ( !Boolean.parseBoolean(ack)) {
+            return null;
+        }
+
+        return getConfig().getTtdWhenAckRequired();
+    }
+
+    @Override
+    protected boolean isCommandValid(final Command command, final Span span) {
+
+        if (!super.isCommandValid(command, span)) {
+            return false;
+        }
+
+        final String conentType = command.getContentType();
+        if (!CONTENT_TYPE_OCTET_STREAM.equals(conentType)) {
```

Comment:
does SigFox require this particular content type?

Reply:
Yes. The command/downstream format is actually pretty strict. It has to be exactly 8 bytes.

Reply:
Then I would suggest to simply ignore whatever has been set on the command AMQP message and just always use this content type when forwarding to the SigFox provider.

Reply:
Well … I wanted to ensure that users are aware of the requirements. Assuming someone would just stick e.g. JSON in there, having 8 bytes by accident, it would work, and the next message (not having 8 bytes) would not. In this case it is clear: 8 bytes of binary data.

And actually the Sigfox backend expects a very specific JSON structure. Which will wrap the payload for the device.

If you think there is a benefit in ignoring the content type, then I will do that. But from a "setting the right expectations", it seemed like a good idea.

Reply:
Not sure if I am getting it correctly.
I understand that the payload is expected to be exactly 8 bytes, right? You are already verifying this, so either the payload is ok regarding size, or it is not. This doesn't seem to be related to whether the payload bytes represent a JSON string or something else, or is it?
My understanding is also that the SigFox adapter will expect to receive the (raw) payload bytes wrapped in a specific JSON structure. However, that JSON structure is created by the adapter, not the application that has sent the (raw) payload bytes in the body of a command, right?
So, the adapter will *always* take the (raw) bytes from the body of the command and (base64 ?) encode them and put them in the JSON structure it sends to the SigFox back end, right?

Reply:
> So, the adapter will always take the (raw) bytes from the body of the command and (base64 ?) encode them and put them in the JSON structure it sends to the SigFox back end, right?

It is hex encoded, but yes.

Reply:
Then I do not really see the issue that you describe with the content type. The payload is *always* 8 bytes, regardless of whether it represents bit vectors of JSON. The receiver of the data will need to know how to interpret this data. The content-type might give a clue to the receiver regarding the structure and encoding of the payload data. However, SigFox doesn't seem to allow you to provide a more specific content type than *it's just bytes*. So, FMPOV, there is nothing wrong with always setting the same content type expected by SigFox in this case.

A sender of a command will need to be aware of the fact that it talks to a SigFox device in any case, I guess ...

Reply:
I simply dropped the check, as it seems that is important to you.